server/src/main/java/io/crate/analyze/AlterTableDropColumnAnalyzer.java

```diff
@@ -0,0 +1,124 @@
+        if (!alterTable.table().partitionProperties().isEmpty()) {
+            throw new UnsupportedOperationException("Dropping a column from a single partition is not supported");
+        }
+
+        DocTableInfo tableInfo = (DocTableInfo) schemas.resolveTableInfo(
+            alterTable.table().getName(),
+            Operation.ALTER,
+            txnCtx.sessionSettings().sessionUser(),
+            txnCtx.sessionSettings().searchPath());
+
+        var exprAnalyzerWithFieldsAsString = new ExpressionAnalyzer(
+            txnCtx, nodeCtx, paramTypeHints, FieldProvider.TO_LITERAL_VALIDATE_NAME, null);
+        var exprCtx = new ExpressionAnalysisContext(txnCtx.sessionSettings());
+
+        List<DropColumnDefinition<Expression>> tableElements = alterTable.tableElements();
+
+        List<DropColumnDefinition<Symbol>> addColumnDefinitions = tableElements.stream().map(tableElement -> new DropColumnDefinition<>(
+            exprAnalyzerWithFieldsAsString.convert(tableElement.name(), exprCtx),
+            tableElement.ifExists()
+        )).toList();
+
+        AnalyzedTableElements<Symbol> analyzedTableElements = TableElementsAnalyzer.analyze(
+            List.copyOf(addColumnDefinitions), tableInfo.ident(), tableInfo, false);
+
+        validate(tableInfo, analyzedTableElements);
+        return new AnalyzedAlterTableDropColumn(tableInfo, analyzedTableElements);
+    }
+
+    private static void validate(DocTableInfo tableInfo, AnalyzedTableElements<Symbol> analyzedTableElements) {
+        for (var colIdentToDrop : analyzedTableElements.columnIdents()) {
+            var refToDrop = tableInfo.getReference(colIdentToDrop);
+            if (refToDrop == null) {
+                throw new ColumnUnknownException(colIdentToDrop, tableInfo.ident());
+            }
+
+            if (tableInfo.primaryKey().contains(refToDrop.column())) {
+                throw new UnsupportedOperationException("Dropping column: \"" + colIdentToDrop.sqlFqn() + "\" which " +
+                                                        "is part of the PRIMARY KEY is not allowed");
+            }
+
+            if (tableInfo.clusteredBy().equals(refToDrop.column())) {
+                throw new UnsupportedOperationException("Dropping column: \"" + colIdentToDrop.sqlFqn() + "\" which " +
+                                                        "is used in 'CLUSTERED BY' is not allowed");
+            }
+
+            if (tableInfo.isPartitioned() && tableInfo.partitionedBy().contains(colIdentToDrop)) {
+                throw new UnsupportedOperationException("Dropping column: \"" + colIdentToDrop.sqlFqn() + "\" which " +
+                                                        "is part of the 'PARTITIONED BY' columns is not allowed");
+            }
+
+            var generatedColRefs = new HashSet<>();
+            for (var genRef : tableInfo.generatedColumns()) {
+                generatedColRefs.addAll(genRef.referencedReferences());
+            }
```

Comment:
Set can be prepared once outside of the loop.

Reply:
thx!